io-engine/tests/snapshot_lvol.rs

```diff
@@ -911,3 +911,178 @@ async fn test_snapshot_volume_provisioning_mode() {
+    let ms = get_ms();
+
+    const LVOL_NAME: &str = "lvol11";
+    const LVOL_SIZE: u64 = 24 * 1024 * 1024;
+    ms.spawn(async move {
+        // Create a pool and lvol.
+        let pool = create_test_pool(
+            "pool11",
+            "malloc:///disk11?size_mb=64".to_string(),
+        )
+        .await;
+        let cluster_size = pool.blob_cluster_size();
+        let lvol = pool
+            .create_lvol(
+                LVOL_NAME,
+                LVOL_SIZE,
+                Some(&Uuid::new_v4().to_string()),
+                true,
+            )
+            .await
+            .expect("Failed to create test lvol");
+
+        // Create a snapshot-1
+        let entity_id = String::from("lvol11_e1");
+        let parent_id = lvol.uuid();
+        let txn_id = Uuid::new_v4().to_string();
+        let snap_name = String::from("lvol11_snap1");
+        let snapshot_uuid = Uuid::new_v4().to_string();
+
+        let snapshot_params = SnapshotParams::new(
+            Some(entity_id),
+            Some(parent_id),
+            Some(txn_id),
+            Some(snap_name),
+            Some(snapshot_uuid),
+            Some(Utc::now().to_string()),
+        );
+        lvol.create_snapshot(snapshot_params.clone())
+            .await
+            .expect("Failed to create a snapshot");
+        // Original Lvol Size to be 0 after snapshot is created.
+        assert_eq!(
+            lvol.usage().allocated_bytes,
+            0,
+            "Volume still has some space allocated after taking a snapshot"
+        );
+        // Write some data to original lvol.
+        bdev_io::write_some(LVOL_NAME, 0, 16, 0xccu8)
+            .await
+            .expect("Failed to write data to volume");
+
+        assert_eq!(
+            lvol.usage().allocated_bytes,
+            cluster_size,
+            "Volume still has some space allocated after taking a snapshot"
+        );
+
+        // Create a snapshot-2 after io done to lvol.
+        let entity_id = String::from("lvol11_e2");
+        let parent_id = lvol.uuid();
+        let txn_id = Uuid::new_v4().to_string();
+        let snap_name = String::from("lvol11_snap2");
+        let snapshot_uuid = Uuid::new_v4().to_string();
+
+        let snapshot_params = SnapshotParams::new(
+            Some(entity_id),
+            Some(parent_id),
+            Some(txn_id),
+            Some(snap_name),
+            Some(snapshot_uuid),
+            Some(Utc::now().to_string()),
+        );
+        lvol.create_snapshot(snapshot_params.clone())
+            .await
+            .expect("Failed to create a snapshot");
+
+        // Original Lvol Size to be 0 after snapshot is created.
+        assert_eq!(
+            lvol.usage().allocated_bytes,
+            0,
+            "Volume still has some space allocated after taking a snapshot"
+        );
+    })
+    .await;
+}
+
+#[tokio::test]
+async fn test_thik_provision_lvol_alloc_after_snapshot() {
```

Comment:
Looks like 'test_thik_provision_lvol_alloc_after_snapshot()' and 'test_thin_provision_lvol_alloc_after_snapshot()' are fully identical (the only difference is flag that denotes thin/thick provisioned test volume) ?
If so, can we have just one function that performs test and just call it from these 2 testcases ? Simethong like this:

```
async fn test_lvol_alloc_after_snapshot(thin: bool) {
   // Do all test logic here.
}

#[tokio::test]
async fn test_thin_provision_lvol_alloc_after_snapshot() {
   test_lvol_alloc_after_snapshot(true).await;
}

#[tokio::test]
async fn test_thick_provision_lvol_alloc_after_snapshot() {
   test_lvol_alloc_after_snapshot(false).await;
}
```

Reply:
Fixed the changes.